core/app/models/spree/variant.rb

```diff
@@ -276,7 +276,7 @@ def set_master_out_of_stock
     # Ensures a new variant takes the product master price when price is not supplied
     def check_price
       if price.nil? && Spree::Config[:require_master_price]
-        raise 'No master variant found to infer price' unless product && product.master
+        errors.add(:base, :no_master_variant_found_to_infer_price) && (return false) unless product && product.master
         raise 'Must supply price for variant or master.price for product.' if self == product.master
```

Comment:
I suppose it'd be a good idea to change `raise` here as well.

Reply:
make sense